app/questionnaire/navigation.py

```diff
@@ -141,7 +141,7 @@ def _generate_repeated_items(self, link_names, first_location, completed_id, no_
             location.group_instance = i
             is_current_group_instance = is_current_group and i == current_group_instance
 
-            if link_names:
+            if link_names and link_names.get(i):
```

Comment:
Is this not just sticking a plaster rather than fixing the issue? why do we have a repeat if it doesn't have a name?